src/main/scala/com/fulcrumgenomics/bam/IdentifyPrimers.scala

```diff
@@ -0,0 +1,1027 @@
+    |
+    |## Fragment Matching (5' and 3')
+    |
+    |The 3' end of the read can also be searched to identify primer pairs using the `--three-prime` option.  In this
+    |case the same algorithm will be employed as paired-end matching for the 5' only case, where each end of the fragment
+    |read is equivalent to the 5' end of the mates of the paired end rad.
+    |
+    |## Unmapped Data
+    |
+    |If no reference is given or mapping information is not available, matching using mapping locations is skipped.
+    |
+    |## Speeding Up Alignment
+    |
+    |Install [the ksw executable](https://github.com/nh13/ksw) manually or via conda (`conda install -c bioconda ksw`)
+    |and supply the path to the `ksw` executable via `--ksw`.
+  """)
+class IdentifyPrimers
+(@arg(flag='i', doc="Input BAM file.")  val input: PathToBam,
+ @arg(flag='p', doc="File containing information about the primers.") val primers: FilePath,
+ @arg(flag='o', doc="Output BAM file.") val output: PathToBam,
+ @arg(flag='m', doc="Path prefix for the metrics files.") val metrics: PathPrefix,
+ @arg(flag='S', doc="Match to primer locations +/- this many bases.") val slop: Int = 5,
+ @arg(          doc="Search for primers on the 3' end of each read as well (5' only by default).") val threePrime: Boolean = false,
+ @arg(          doc="Maximum mismatches for a primer to be considered a match.") val maxMismatches: Int = 1,
+ @arg(          doc="The minimum alignment score for fully gapped alignment for a primer to be considered a match.") val minAlignmentScore: Int = 10,
+ @arg(          doc="The match score to use for aligning to primer sequences.") val matchScore: Int = 1,
+ @arg(          doc="The mismatch score to use for aligning to primer sequences.") val mismatchScore: Int = -4,
+ @arg(          doc="The gap open score to use for aligning to primer sequences.") val gapOpen: Int = -6,
+ @arg(          doc="The gap extension score to use for aligning to primer sequences.") val gapExtend: Int = -1,
+ @arg(          doc="The minimum difference in mismatches between the best match and second best match to accept a primer match.  Not used for full alignment.") val minMismatchDelta: Int = 1,
+ @arg(flag='t', doc="The number of threads to use.") val threads: Int = 1,
+ @arg(flag='n', doc="Examine the first N templates.") val numTemplates: Option[Int] = None,
+ @arg(          doc="Skip full-alignment matching") val skipFullAlignment: Boolean = false,
+ @arg(          doc="Path to the ksw aligner.") val ksw: Option[String] = None,
+ @arg(          doc="The maximum number of templates in memory.") val maxTemplatesInRam: Option[Int] = None,
+ @arg(          doc="The number of templates to process at a time per thread.") val templatesPerThread: Int = 1000
+) extends FgBioTool with LazyLogging {
+
+  import IdentifyPrimers._
+
+  private val kswExecutable: Option[FilePath] = this.ksw.map(Paths.get(_)).flatMap {
+    case p if Files.exists(p) => Some(p)
+    case name =>
```

Comment:
this branch will also get triggered if the user typos, e.g. `/usr/locla/bin/ksw`, and that's going to be very confusing.  Could you instead just try executing the executable given and see if it works?